### DISCUSSION TITLE
[Feature] document how dates are determined for photos / videos in timeline

### DISCUSSION THREAD

**Discussion by twitsforbrains** [immich-app/immich#1947] [Feature Request]
### The feature

I used the bulk import tool to import my existing photos. However a bunch of them had incorrect dates. I don't see any documentation on what data is used to determine the date for these photos. It would be fairly easy to simply add a paragraph in the faq so that users who want to, can use tools like exiftool to update or use 'touch' to modify create / modify times depending on the logic.

So I guess my question is, what data is used for photo/video times? (I can make a PR if this is needed)

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by alextran1502** [comment]
The mechanism we use to determine the date is like this

1. When first backup from any client, the date will be the creation date of the file
2. After that, it will go through the pipeline of metadata extraction and then use the EXIF creation date of the file for a more accurate result.
3. If none are available, then it will fall back to the modified date.

**Comment by twitsforbrains** [comment]
I created a [PR here](https://github.com/immich-app/immich/pull/1978).

However before we merge, can you clarify a few things for me? 

You first mention the Creation date of the file in step 1, but then say it falls back to the Modified date in step 3. Linux / Unix only has Access Times and Modified Times. I assume you mean the modified time in step 1?

Regarding step 2, which exif metadata is used? I know there is CreateDate and DateTimeOriginal that is commonly used but it's not clear to me which one should I update.

Finally, I was wondering if the files are already uploaded, and I update the exif metadata within the storage, is there a way to update the photo in the timeline ? (I.e. can I run a job of some sort?) I think it would be helpful to add this detail if it is possible since it would give people a way to edit the date of a photo without needing to delete / reupload the file. (If its not possible that's fine too, but I would like to document it so that users know they should make sure the files have metadata before uploading).

PS sorry for creating a PR and not a draft.

**Reply by alextran1502** [reply]
Here is the library that we used to extract the EXIF date-time information, it based on a few heuristics to determine the best possible timezone

https://photostructure.github.io/exiftool-vendored.js/index.html#dates

If the files are already uploaded and you update the EXIF metadata, you can run the metadata extraction job again to initiate the extraction process. The caveat here is that even if you only change one file, you have to run the extraction job for all assets.
